python/tank/deploy/git_descriptor.py

```diff
@@ -259,8 +188,95 @@ def _find_latest_by_pattern(self, pattern):
             new_loc_dict,
             self._type
         )
-        
 
+    def _find_latest_tag_by_pattern(self, version_numbers, pattern):
+        """
+        Given a list of version numbers, find the one that best matches the given pattern.
```

Comment:
Perhaps to avoid any confusion, we should think about referring to these as version strings throughout instead of version numbers?